docs/using-react-redux/static-types.md

```diff
@@ -0,0 +1,106 @@
+---
+id: static-types
+title: Type safety and React-Redux
+hide_title: true
+sidebar_label: Type safety and React-Redux
+---
+
+# Type safety and React-Redux
+
+The [`react-redux` type definitions](https://npm.im/@types/react-redux) export some helpers to make it easier to write typesafe interfaces between your redux store and your React components.
+
+## Typing the useSelector hook
+
+If you manually type your selector functions, `useSelector` will return the same type as your selector
+
+```ts
+// selectIsOn always returns a boolean, so isOn is typed as a boolean
+const isOn = useSelector(selectIsOn)
+```
+
+Otherwise `useSelector` requires you to manually type the state argument every time you use it.
+
+```ts
+const isOn = useSelector((state: MyStateType) => state.isOn)
+```
+
+If you'd like, you can define a typed `useSelect` hook using a helper type
+
+```ts
+// reducer.ts
+import { useSelector, TypedUseSelectorHook } from 'react-redux'
+
+interface MyStateType {
+  isOn: boolean
+}
+
+export const useTypedSelector: TypedUseSelectorHook<MyStateType> = useSelector
+
+// my-component.tsx
+import { useTypedSelector } from './reducer.ts'
+
+const isOn = useSelector(state => state.isOn)
+```
+
+## Typing the `connect` higher order component
+
+React-redux exposes a helper type, `ConnectedProps`, that can extract the return type of mapStateToProps and mapDispatchToProps.
+
+```ts
+import { connect, ConnectedProps } from 'react-redux'
+
+interface MyStateType {
+  isOn: boolean
+}
+
+const connector = connect(
```

Comment:
I'd want to highlight that normally we do the `connect(mapState, mapDispatch)(MyComponent)` double-function-call all in one step at the end, but for this to work we break it up into two steps (and particularly doing the first part _before_ the component definition) so we can infer the props from the `connector` function.